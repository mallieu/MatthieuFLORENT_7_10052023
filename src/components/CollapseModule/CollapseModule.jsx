import { useState } from 'react';
import { ChevronToggle } from '../../assets/images/Chevron-toggle';

const CollapseModule = ({ title, content }) => {
    const [stateModule, setStateModule] = useState(true);

    function CollapseModuleClick() {
        setStateModule(!stateModule);
    }
    return (
        <>
            <div className="collapseModule">
                <div className="collapseModule__titleBanner">
                    <h4>{title}</h4>
                    <ChevronToggle
                        className="collapseModule__toggle"
                        state={stateModule}
                        onClick={CollapseModuleClick}
                    />
                </div>
                {stateModule && (
                    <div className="collapseModule__content">{content}</div>
                )}
            </div>
        </>
    );
};

export default CollapseModule;

// {stateModule && (
//     <div
//         className={`collapseModule__content
//         ${
//             stateModule
//                 ? 'collapseModule__content--collapsed'
//                 : 'collapseModule__content--uncollapsed'
//         }
//          `}
//     >
