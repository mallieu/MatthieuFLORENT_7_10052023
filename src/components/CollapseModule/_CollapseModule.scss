.collapseModule {
    font-size: map-get($taille-texte, m);

    &__titleBanner {
        align-items: center;
        background-color: $couleur--principale;
        border-radius: $radius-s;
        color: white;
        display: flex;
        max-width: 100%;
        justify-content: space-between;
        padding: 10px 15px;
        position: relative;
        z-index: 1;

        h4 {
            margin: 0;
        }
    }

    &__toggle {
        height: 20px;
        fill: white;
        cursor: pointer;
    }

    &__content {
        border-radius: 0 0 $radius-s $radius-s;
        background-color: $couleur--gris-clair;
        font-weight: map-get($graisse-texte, light);
        line-height: map-get($taille-line-height, l);
        margin-top: -$margin-s;
        padding: 40px 20px 20px;
        position: relative;

        @include mediaqueries-max ($tablet) {
            margin-top: -$margin-m;
        }

        ul {
            padding-left: 0;
            margin: 0;
            font-size: map-get($taille-texte, m);
            list-style: none;
        }
    }
}