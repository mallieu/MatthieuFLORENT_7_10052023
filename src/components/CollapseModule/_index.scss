.collapseModule {
    &__title {
        max-width: 100%;
        padding: 10px 15px;
        background-color: $couleur--principale;
        border-radius: $radius-s;
        color: white;
        font-weight: map-get($graisse-texte, light);
    }

    &__content {
        padding: 10px;
        background-color: $couleur--gris-clair;
        border-radius: $radius-s;

    }
}