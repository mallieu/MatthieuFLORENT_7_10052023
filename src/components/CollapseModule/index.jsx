const CollapseModule = ({ title, content }) => {
    return (
        <>
            <div className="collapseModule">
                <h4 className="collapseModule__title">{title}</h4>
                <div className="collapseModule__content">{content}</div>
            </div>
        </>
    );
};

export default CollapseModule;
