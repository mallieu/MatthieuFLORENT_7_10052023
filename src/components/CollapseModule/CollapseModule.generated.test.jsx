import renderer from 'react-test-renderer';
import { useState } from 'react';
import CollapseModuleToggle from '../../assets/icons/CollapseModuleToggle';
import CollapseModule from './CollapseModule';

jest.mock('../../assets/icons/CollapseModuleToggle');

const renderTree = (tree) => renderer.create(tree);
describe('<CollapseModule>', () => {
    it('should render component', () => {
        expect(renderTree(<CollapseModule />).toJSON()).toMatchSnapshot();
    });
    it('should render component with props', () => {
        expect(
            renderTree(
                <CollapseModule
                    title={'Mon test'}
                    content={'Mon contenu de test'}
                />
            ).toJSON()
        ).toMatchSnapshot();
    });
});
