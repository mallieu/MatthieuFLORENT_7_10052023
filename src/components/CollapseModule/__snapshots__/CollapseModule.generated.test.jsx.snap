// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseModule> should render component 1`] = `
<div
  className="collapseModule__container"
>
  <div
    className="collapseModule__titleBanner"
  >
    <h4 />
  </div>
  <div
    className="collapseModule__content"
  />
</div>
`;

exports[`<CollapseModule> should render component with props 1`] = `
<div
  className="collapseModule__container"
>
  <div
    className="collapseModule__titleBanner"
  >
    <h4>
      Mon test
    </h4>
  </div>
  <div
    className="collapseModule__content"
  >
    Mon contenu de test
  </div>
</div>
`;
